src/libs/DistanceRequestUtils.ts

```diff
@@ -38,6 +39,49 @@ Onyx.connect({
 const METERS_TO_KM = 0.001; // 1 kilometer is 1000 meters
 const METERS_TO_MILES = 0.000621371; // There are approximately 0.000621371 miles in a meter
 
+/**
+ * Retrieves the mileage rates for given policy.
+ *
+ * @param policy - The policy from which to extract the mileage rates.
+ * @param includeDisableRate - Should include disable rate
+ *
+ * @returns An array of mileage rates or an empty array if not found.
+ */
+function getMileageRates(policy: OnyxEntry<Policy>, includeDisableRate = true): Record<string, MileageRate> {
+    const mileageRates: Record<string, MileageRate> = {};
+
+    if (!policy || !policy?.customUnits) {
+        return mileageRates;
+    }
+
+    const distanceUnit = Object.values(policy.customUnits).find((unit) => unit.name === CONST.CUSTOM_UNITS.NAME_DISTANCE);
+    if (!distanceUnit?.rates) {
+        return mileageRates;
+    }
+
+    Object.entries(distanceUnit.rates).forEach(([rateID, rate]) => {
+        if (includeDisableRate) {
+            mileageRates[rateID] = {
+                rate: rate.rate,
+                currency: rate.currency,
+                unit: distanceUnit.attributes.unit,
+                name: rate.name,
+                customUnitRateID: rate.customUnitRateID,
+            };
+        } else if (rate.enabled) {
+            mileageRates[rateID] = {
+                rate: rate.rate,
+                currency: rate.currency,
+                unit: distanceUnit.attributes.unit,
+                name: rate.name,
+                customUnitRateID: rate.customUnitRateID,
+            };
+        }
```

Comment:
@cretadn22 Let's DRY up the logic here!

```suggestion
        if (!includeDisableRate && !rate.enabled) {
            return;
        }
    
        mileageRates[rateID] = {
            rate: rate.rate,
            currency: rate.currency,
            unit: distanceUnit.attributes.unit,
            name: rate.name,
            customUnitRateID: rate.customUnitRateID,
        };
```